runtime/float.go

```diff
@@ -28,12 +29,16 @@ var FloatType = newBasisType("float", reflect.TypeOf(Float{}), toFloatUnsafe, Ob
 // Float represents Python 'float' objects.
 type Float struct {
 	Object
-	value float64
+	value    float64
+	hashOnce sync.Once
```

Comment:
Once is heavyweight for this purpose since it relies on a mutex. I would recommend using an approach similar to what we do for str using atomic pointer operations.